language/ast.py

```diff
@@ -118,7 +119,7 @@ class ParenExpression(Expr):
     data = Field(type=Expr)
 
 
-class RefinementSpec(Type):
+class RefinementSpec(Declaration):
     """
     Refinement specification (for Message use (Field => Inner_Type))
     """
```

Comment:
There are still inconsistencies in the naming, e.g. `TypeDecl(Declaration)` vs. `RefinementSpec(Declaration)`. I would rename everything named `*Spec` to `*Decl`:

```
language/ast.py:class RefinementSpec(Declaration):
language/ast.py:class SessionSpec(Declaration):
language/ast.py:class PackageSpec(RFLXNode):
```

`PackageSpec` could also just be `Package`.

Reply:
Changed in dce744528c85e4b63b5b86f2fd40b00ddd8233a9.